flowapi/flowapi/user_model.py

```diff
@@ -24,26 +23,31 @@ class UserObject:
     ----------
     username : str
         Name of the user
-    claims : dict
+    scopes : dict
         Dictionary giving a whitelist of the user's claims
     """
 
-    def __init__(self, username: str, scopes: List[str]) -> None:
+    def __init__(self, username: str, scopes: dict) -> None:
         self.username = username
         self.scopes = scopes
 
     def has_access(self, *, actions: List[str], query_json: dict) -> bool:
-
+        breakpoint()
         try:
-            scopes = set(query_to_scope_list(query_json))
+            claims = set(query_to_scopes(query_json))
         except Exception as exc:
             raise BadQueryError
         if "query_kind" not in query_json:
             raise MissingQueryKindError
-
-        for action in actions:
-            if {*scopes, action} in self.scopes:
-                return True
+        granting_roles = []
+        for role, scopes in self.scopes.items():
+            if all(x in scopes for x in {*claims, *actions}):
```

Comment:
I'm wondering if `get_result` should imply `can_poll`, or if we should just implement an explicit `poll` scope - it wouldn't be that much extra work to do. I can see one niche for it, if we have some kind of automatic provisioning or resource role that needs to keep an eye on what queries are running, but doesn't care what they are.

Reply:
I think we want 'run' and 'get_result' to both imply 'poll', because in either of those situations you might need to be able to check the query state. I think the best solution is probably to just require the appropriate query scopes to poll a query, rather than requiring the query scopes _and_ one of ('run', 'get_result').

Reply:
Yeah, that makes sense - I'll add it to the list

Reply:
In branch + tested with `test_granular_poll_access`